src/classes/presentation_controller.vala

```diff
@@ -322,38 +353,114 @@ namespace pdfpc {
         private double pointer_x;
         private double pointer_y;
 
+        public bool is_eraser_active() {
+            return drawing_enabled && current_mouse_tool == overlay_drawing.eraser;
+        }
+
+        public bool is_pen_active() {
+            return drawing_enabled && current_mouse_tool == overlay_drawing.pen;
+        }
+
+        public bool is_pointer_active() {
+            return pointer_enabled;
+        }
+
+        public void toggle_eraser() {
+            if (!mouse_tool_is_eraser) {
+                current_mouse_tool = overlay_drawing.eraser;
+            } else {
+                current_mouse_tool = overlay_drawing.pen;
+            }
+            mouse_tool_is_eraser = !mouse_tool_is_eraser;
+            // don't allow drawing to continue
+            drawing_have_last = false;
+            this.controllables_update();
+        }
+
+        public void increase_drawing_pen() {
+            if (current_drawing_tool.width < 500) current_drawing_tool.width += drawing_step;
+            queue_surface_draws();
+        }
+
+        public void decrease_drawing_pen() {
+            if (current_drawing_tool.width > drawing_step) current_drawing_tool.width -= drawing_step;
+            queue_surface_draws();
+        }
+
+        private void hide_or_show_surfaces() {
+            if (pointer_enabled || drawing_present) {
+                if (presenter!=null) presenter_surface.show();
+                if (presentation!=null) presentation_surface.show();
+            } else {
+                if (presenter!=null) presenter_surface.hide();
+                if (presentation!=null) presentation_surface.hide();
+            }
+        }
+
+        private void queue_surface_draws() {
+            if (presenter!=null) presenter_surface.queue_draw();
+            if (presentation!=null) presentation_surface.queue_draw();
+        }
+
         protected void init_presentation_pointer(Gtk.Allocation a) {
             presentation_allocation = a;
             presentation_surface = new Gtk.DrawingArea();
             presentation_surface.set_size_request(a.width, a.height);
+            overlay_drawing = Drawings.create(metadata, a.width, a.height);
```

Comment:
Drawing and pointing is (for me) unrelated. Could you put this in a separate function.